src/modules/RA/DataLayer/components/tabs/StyleTab/Style/WizardIcon.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import { useTranslate, RadioButtonGroupInput } from 'react-admin';
+import { makeStyles } from '@material-ui/core/styles';
+import FormLabel from '@material-ui/core/FormLabel';
+
+import useSettings from '../../../../../../../hooks/useSettings';
+import TextStyleField from './TextStyleField';
+import SizeStyleField from './SizeStyleField';
+
+import styles from './styles';
+
+const useStyles = makeStyles(styles);
+
+const defaultSpriteBaseUrl = 'https://makina-icon.netlify.app/sprite';
+
+const WizardIcon = ({ path, fields, getValuesOfProperty }) => {
+  const classes = useStyles();
+  const translate = useTranslate();
+  const [iconChoices, setIconChoices] = React.useState([]);
+  const { spriteBaseUrl = defaultSpriteBaseUrl } = useSettings();
+
+  React.useState(() => {
+    let mounted = true;
+    const loadIconChoices = async () => {
+      let choices;
+      try {
+        const result = await fetch(`${spriteBaseUrl}.json`);
+        const data = await result.json();
```

Comment:
I'm not sure if your code or the following is more optimized but it's more concise with the `then` chaining
```js
const data = await fetch(`${spriteBaseUrl}.json`).then(result => result.json());
```

Reply:
I think perf are exactly the same. I propose another more concise and consistent version:

```js
const data = await (await fetch(`${spriteBaseUrl}.json`)).json();
```